src/main/java/com/hotels/bdp/circustrain/bigquery/metastore/BigQueryMetastoreClient.java

```diff
@@ -131,6 +131,7 @@ public Table getTable(String databaseName, String tableName) throws TException {
     log.info("Getting table {}.{} from BigQuery", databaseName, tableName);
     checkDbExists(databaseName);
     com.google.cloud.bigquery.Table table = getBigQueryTable(databaseName, tableName);
+    dataExtractionManager.extract(table);
```

Comment:
hmm I know you got a cache in the dataExtractionManager so calling this multiple times won't actually trigger another extraction but it still feels a bit wrong doing this here. A method called getTable should't get a table to run an extraction. You might still need to explore the Copier a bit and add your own BigQuery copier so you have full control.

Reply:
I dunno, although this is a bit of a kludge that's adding a lot of code to do something so simple, and also adding a dependency on every copier implementation but S3-S3 Copier, plus some not so elegant code to do the delegation to the copiers and also some weird stuff with Spring copier ordering, I think the end result will likely be worse than calling a method in a not so ideal place

Reply:
if CT doesn't allow for an elegant way to add a copier, than that's something we need to address in CT. I agree that it should not imply having dependencies on all the copiers.